index.js

```diff
@@ -24,13 +24,11 @@ module.exports = {
     this.modulesPreprocessor = new ModulesPreprocessor({ owner: this });
     this.outputStylesPreprocessor = new OutputStylesPreprocessor({ owner: this });
     this.checker = new VersionChecker(this.project);
+    this.plugins = new PluginRegistry(this.parent);
```

Comment:
This is potentially a breaking change.

We have an addon like this:

```js
module.exports = {
  name: require('./package').name,

  createCssModulesPlugin(parent) {
    return new BreakpointsPlugin(parent, {
      breakpoints: this.breakpoints
    });
  },

  included() {
    this.breakpoints = require('./breakpoints.json');
  }
}
```

Prior to this change, `included` was called before `createCssModulesPlugin`. Now the order is changed.

For this simple case, the fix is easy:

```js
module.exports = {
  name: require('./package').name,

  createCssModulesPlugin(parent) {
    return new BreakpointsPlugin(parent, {
      breakpoints: this.breakpoints
    });
  },

  init(...args) {
    this._super(...args);
    this.breakpoints = require('./breakpoints.json');
  }
}
```

Reply:
☹️ My goal was for this overall release to be non-breaking, but I don't see a way around this timing change, because we now need access to the ECM options before we can set up the HTMLBars plugin. Technically we never made any promises about invocation order, but hiding behind that rules-lawyering feels pretty bad.

On the public registry, at least, I think you and I are the only two authors of packages with the [`ember-css-modules-plugin` keyword](https://www.npmjs.com/search?q=keywords:ember-css-modules-plugin), though of course there's no way to know about private packages. Do you have any sense of how much this is likely to impact the plugins you've written? I don't _think_ any of mine (internal or public) are relying on it, but they're mostly fairly simple.

Reply:
Personally and speaking for @ClarkSource I'm totally fine with this change. None of my public plugins break, and of all private ones this was the only one. 🎉